### PR TITLE
RELEASE: 1.6.1

### PR DESCRIPTION
**Pull request by jamesmcgill**
### Description

Release v1.6.1 with fix for compiler errors in 2022.1 editors and with XRI toolkit

### Changes made

Bump to v1.6.1
Hide Experimental UITK code completely behind compiler define

### Checklist

Before review:

- [ ] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [ ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

### CODE REVIEW

**Review by jfreire-unity — APPROVED** [review]

File: Assets/Tests/InputSystem.Editor/TestData.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#if UNITY_2022_2_OR_NEWER
+#if UNITY_INPUT_SYSTEM_UI_TK_ASSET_EDITOR
```
Comment (lyndon-unity) [inline Assets/Tests/InputSystem.Editor/TestData.cs:1]: I'm assuming this is UI TK specific but its not 100% obvious to me.

File: Assets/Tests/InputSystem.Editor/TestDataGenerators.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#if UNITY_2022_2_OR_NEWER
+#if UNITY_INPUT_SYSTEM_UI_TK_ASSET_EDITOR
```
Comment (lyndon-unity) [inline Assets/Tests/InputSystem.Editor/TestDataGenerators.cs:1]: I'm assuming this is UI TK specific but its not 100% obvious to me.

File: Packages/com.unity.inputsystem/InputSystem/Actions/InputControlScheme.cs [hunk 3]
```diff
@@ -112,7 +112,7 @@ public InputControlScheme(string name, IEnumerable<DeviceRequirement> devices =
             }
         }
 
-        #if UNITY_EDITOR && UNITY_2022_2_OR_NEWER
+        #if UNITY_EDITOR && UNITY_INPUT_SYSTEM_UI_TK_ASSET_EDITOR
         internal InputControlScheme(SerializedProperty sp)
```
Comment (lyndon-unity) [inline Packages/com.unity.inputsystem/InputSystem/Actions/InputControlScheme.cs:116]: I'm assuming this is UI TK specific but its not 100% obvious to me.

**Review by lyndon-unity — APPROVED** [review]
Overall this looks sensible, couple of parts removed that seem to make sense but not 100% clear UI TK related (just due to my lack of familiarity)

**Comment by jamesmcgill** [comment]
All defines changed in this PR were added as part of the UITK merge: https://github.com/Unity-Technologies/InputSystem/pull/1668

I checked them individually beforehand. Whether they really needed guards is an issue for another day.

**Review by lyndon-unity — APPROVED** [review]
Seems required.

**Review by lyndon-unity — APPROVED** [review]
Seems required.

**Pull request merged**